dynparquet/row.go

```diff
@@ -10,12 +12,53 @@ type DynamicRow struct {
 	DynamicColumns map[string][]string
 }
 
-func (s *Schema) RowLessThan(a, b *DynamicRow) bool {
+// Intersection returns the intersection of two schemas
+// This is useful when comparing two rows of different schemas to only compare the columns that intersect.
+func Intersection(a, b *parquet.Schema) *parquet.Schema {
```

Comment:
This seems like a lot of trouble to go through to just use `.ChildNames()` directly on the result in the only place where this function is used. Why not just merge the two string slices?